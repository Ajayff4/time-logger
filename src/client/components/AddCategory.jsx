import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import axios from 'axios'
import { addCategory } from '../actions';
import { connect } from 'react-redux';

class AddCategory extends Component {
    constructor() {
        super();
        this.state = {
            category: ""
        }
    }

    goToAddTag = () => {
        this.props.history.push("/add-tag");
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit = (e) => {
        e.preventDefault();
        const { category } = this.state;
        axios.post('/add-category', {
            category: category
        }).then(res => {
            this.props.addCategory()
            this.props.history.push('/add-log')
        }).catch(err => {
            console.error("post: ", err);
        });
    }

    render() {
        return (
            <>
                <h2>Add Category Component</h2>
                <center>
                    <div id="form-css">
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="category"
                                id="category"
                                placeholder="Input category name"
                                value={this.state.category}
                                onChange={this.onChange}
                            />
                            <br />
                            <button type="submit" value="submit">Add Category</button>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.goToAddTag}>back to add tag</a>
                        </form>
                    </div>
                </center>
            </>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        addCategory: () => dispatch(addCategory())
    }
}

export default connect(null, mapDispatchToProps)(withRouter(AddCategory))