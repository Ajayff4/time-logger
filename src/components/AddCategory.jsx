import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import axios from 'axios'
import * as actions from '../actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

class AddCategory extends Component {
    static propTypes = {
        addCategory: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);
        this.state = {
            category: ""
        }
    }

    goToAddTag = () => {
        this.props.history.push("/add-tag")
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault()
        axios.post('http://localhost:5000/rest/category/add', {
            category: this.state.category
        },{
            headers: {"Content-type": "application/json"}
        })
        .then(res => {
            this.props.addCategory()
            this.props.history.push('/add-tag')
        }).catch(err => {
            this.props.failedToFetchData("Error in add category" + err);
        })
    }

    render() {
        return (
            <>
                <h2>Add Category Component</h2>
                <center>
                    <div id="form-css">
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="category"
                                id="category"
                                placeholder="Input category"
                                value={this.state.category}
                                onChange={this.onChange}
                                required
                            />
                            <br />
                            <button type="submit" value="submit">Add Category</button>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.goToAddTag}>back to add tag</a>
                        </form>
                    </div>
                </center>
            </>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        addCategory: () => dispatch(actions.addCategory()),
        failedToFetchData: (err) => dispatch(actions.failedToFetchData(err))
    }
}

export default connect(null, mapDispatchToProps)(withRouter(AddCategory))