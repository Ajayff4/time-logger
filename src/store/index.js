import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store