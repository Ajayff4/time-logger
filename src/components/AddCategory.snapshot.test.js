import React from 'react'
import renderer from 'react-test-renderer'
import AddCategory from './AddCategory'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('AddCategory Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><AddCategory /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});