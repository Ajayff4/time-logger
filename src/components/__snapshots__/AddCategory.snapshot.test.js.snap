// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Snapshot Testing... renders component correctly 1`] = `
<div
  className="addCategoryComponent"
>
  <h2>
    Add Category Component
  </h2>
  <center>
    <div
      id="form-css"
    >
      <form
        method="POST"
        onSubmit={[Function]}
      >
        <input
          id="category"
          name="category"
          onChange={[Function]}
          placeholder="Input category"
          required={true}
          type="text"
          value=""
        />
        <br />
        <button
          type="submit"
          value="submit"
        >
          Add Category
        </button>
        <br />
        <a
          href="#"
          onClick={[Function]}
        >
          back to add tag
        </a>
      </form>
    </div>
  </center>
</div>
`;
